src/mscorlib/src/System/Diagnostics/Eventing/EventSource.cs

```diff
@@ -5804,6 +5828,74 @@ public enum EventManifestOptions
         AllowEventSourceOverride = 0x8,
     }
 
+#if FEATURE_EVENTSOURCE_XPLAT 
+
+    internal  class XplatEventLogger : EventListener
+    {
+        public XplatEventLogger() {}
+        
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        public static extern bool IsEventSourceLoggingEnabled();
+
+        [DllImport(JitHelpers.QCall, CharSet = CharSet.Unicode)]
+        private static extern void LogEventSource(int eventID, string eventName, string eventSourceName, string payload);
+
+        private static string Serialize(ReadOnlyCollection<string> payloadName, ReadOnlyCollection<object> payload, string sep = ", ")
+        {
+
+            if (payloadName == null || payload == null )
+                return String.Empty;
+
+            if (payloadName.Count == 0 || payload.Count == 0)
+                return String.Empty;
+
+            Contract.Assert(payloadName.Count == payload.Count);
+            
+            var sb = StringBuilderCache.Acquire();
+
+            for (int i = 0; i < payloadName.Count; i++)
```

Comment:
I see.  We should definitely make sure the JSON is valid so that it's simple to parse on the other end.  So, making sure that we have {}, quoted strings, etc. is important.

If you need to test, you should be able to take the results that are generated and just put them into a file and use something like Json.NET to parse them.  How does that sound?